ldap/servers/slapd/mapping_tree.c

```diff
@@ -605,6 +605,37 @@ mapping_tree_entry_add(Slapi_Entry *entry, mapping_tree_node **newnodep)
         return lderr;
     }
 
+    /* Verify there is a matching backend for this suffix */
```

Comment:
Actually, I forgot a point here. When we are doing the check of the BE, we should either assert/validate that the suffix from the MT == BE suffix *or*, we should actually just ignore the suffix from the MT config entry, and we should rely on the suffix value from the BE instead. What do you think?

Reply:
> Actually, I forgot a point here. When we are doing the check of the BE, we should either assert/validate that the suffix from the MT == BE suffix 

Won't that accomplish the exact same thing this code is already doing?

> or, we should actually just ignore the suffix from the MT config entry, and we should rely on the suffix value from the BE instead. What do you think?  

Why would we ignore the invalid configuration in the MT entry?  Sorry I am probably missing your point :-/

Reply:
When we create the MT node, we use the subtree (suffix) from get_subtree_from_entry which uses the value of cn in the mapping tree. But we never check that cn == backend suffix.

So with your change, we now assert that a backend for MT nsslapd-backend also exists, but we never validate that the backend suffix == mt suffix.

IMO we should either check these are the same value *or* even better, we only read the suffix from the BE because the BE knows what suffix it provides, so we remove the possibility of inconsistency from the MT entries.

Reply:
> When we create the MT node, we use the subtree (suffix) from get_subtree_from_entry which uses the value of cn in the mapping tree. But we never check that cn == backend suffix.
> 
> So with your change, we now assert that a backend for MT nsslapd-backend also exists, but we never validate that the backend suffix == mt suffix.

Yes it does:

    slapi_sdn_compare(subtree, be->be_suffix) == 0)

The suffixes must match.  Sorry, I'm still missing your point here :-(

Reply:
With the MT are there types where we don't have an associated backend? Referrals maybe? We probably should just check in the case where the MT has a backend that suffix's are equal I think.

Reply:
Ohhhh okay, I missed that line. IN that case, ignore me, this is all good then :D